### DISCUSSION TITLE
Fame cannot be assigned to the image attribute and inherits the IInputArray interface. Why? 4.8.1.5350

### DISCUSSION THREAD

**Discussion by leitengcn** [emgucv/emgucv#890] [Q&A]
private void Capture_ImageGrabbed(object sender, EventArgs e)
 {
     Mat frame = new Mat();
     try
     {                
         capture.Retrieve(frame, 0);
         
         imagebox1.Image = frame;
         //pictureBox1.Image = frame.ToBitmap();

         if (shot)
         {
             frame.Save(AppDomain.CurrentDomain.SetupInformation.ApplicationBase + $"images\\screenshot_{DateTime.Now.ToString("yyyy-MM-dd_HH-mm-ss")}.jpg");
             shot = false;
         }
         if (recording && videoWriter != null)
         {
             videoWriter.Write(frame);
         }
     }
     catch (Exception ex)
     {

         throw;
     }
     finally
     {
         GC.Collect(); //强制进行垃圾回收
     }
 }

**Comment by leitengcn** [comment]
This code can be read and assigned to image (IInputArray) normally in the old version 3.1.0.1

**Reply by emgucv** [reply]
Do you means an error in this line:
`imagebox1.Image = frame;`

What is the type of "imagebox1"? Is it Emgu.CV.UI.ImageBox?

https://www.emgu.com/wiki/files/4.8.1/document/html/P_Emgu_CV_UI_ImageBox_Image.htm

It works in our demo applications.

**Reply by leitengcn** [reply]
Haven't you encountered it on your end? I can't even use the latest version of the library when doing rtsp streaming

**Comment by leitengcn** [comment]
yes,it's Emgu.CV.UI.ImageBox ,&nbsp;inherits the IInputArray interface


雷腾
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "emgucv/emgucv"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月15日(星期三) 凌晨0:24
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [emgucv/emgucv] Fame cannot be assigned to the image attribute and inherits the IInputArray interface. Why? 4.8.1.5350 (Discussion #890)





 
Do you means an error in this line:
 imagebox1.Image = frame;
 
What is the type of "imagebox1"? Is it Emgu.CV.UI.ImageBox?
 
https://www.emgu.com/wiki/files/4.8.1/document/html/P_Emgu_CV_UI_ImageBox_Image.htm
 
It works in our demo applications.
